liquidhaskell-boot/src/Language/Haskell/Liquid/Transforms/CoreToLogic.hs

```diff
@@ -292,6 +292,7 @@ coreToLg (C.Case e b _ alts)
 -- coreToLg (C.Lam x e)           = do p     <- coreToLg e
 --                                     tce   <- lsEmb <$> getState
 --                                     return $ ELam (symbol x, typeSort tce (GM.expandVarType x)) p
+coreToLg (C.Case e _ _ [])     = coreToLg e
```

Comment:
Looks to me like this might produce a term of the wrong type. The scrutinee doesn't need to have the same type as the case statement.